x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/JdbcResultSetMetaData.java

```diff
@@ -139,7 +139,7 @@ public boolean isDefinitelyWritable(int column) throws SQLException {
 
     @Override
     public String getColumnClassName(int column) throws SQLException {
-        return TypeUtils.classOf(column(column).type).getName();
+        return column(column).isArray ? JdbcArray.class.getName() : TypeUtils.classOf(column(column).type).getName();
```

Comment:
This is incorrect since this method is used to create the value container on the java side.
`int[] -> int[].class.getName() -> [I`, `int[][] -> int[][].class -> [[I`, `Integer[] -> Integer[].class.getName() -> [LInteger;`, etc...

Reply:
Sure, that could also make sense and I could update it inline with these examples, but just for comparison, taking [PostgreSQL's JDBC driver](https://github.com/pgjdbc/pgjdbc) as a comparison, running the following query:
 `SELECT ARRAY[1, 2, 5], ARRAY['foo', 'bar'], 'foobar'`
 will return for the the following methods:
 * `resultSet.getMetaData().getColumnName(i)`
 * `.getColumnTypeName(i)`
 * `.getColumnClassName(i)`
 * `.getColumnType(i)`

the following results:
```
           array|           array|        ?column?|
           _int4|           _text|            text|
  java.sql.Array|  java.sql.Array|java.lang.String|
            2003|            2003|              12|
```

Reply:
It's worth checking other drivers and potentially the JDBC spec to see whether there's consensus on whether the interface name should be returned or the concrete class.
Since there's a chance of nested arrays, returning `java.sql.Array` makes more sense.